### PR TITLE
Ensure same domain when converting geometry.

### PR DESCRIPTION
**Pull request by mgjarrett**
## Description

`initNewReactor` creates a `newReactor` from a `sourceReactor`'s blueprints. Sometimes the `sourceReactor` has been expanded from `THIRD` to `FULL` core geometry, but the blueprints do not reflect this change. This code change ensures that the `newReactor` is also expanded to full core if the `sourceReactor` has been expanded.

## Checklist

- [x] The code is understandable and maintainable to people beyond the author.
- [x] Tests have been added/updated to verify that the new or changed code works.
- [x] There is no commented out code in this PR.
- [x] The commit message follows [good practices](https://terrapower.github.io/armi/developer/tooling.html).
- [x] All docstrings are still up-to-date with these changes.

### CODE REVIEW

**Review by jakehader — APPROVED** [review]

**Comment by jakehader** [comment]
@john-science I'll let you merge.

**Comment by mgjarrett** [comment]
@john-science I think this can be merged if you're good with it.

**Pull request merged**
